### PR TITLE
add wal function

### PR DESCRIPTION
**Pull request by hust-hhb**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/runtime/group_commit_mgr.h [hunk 1]
```diff
@@ -53,6 +54,11 @@ class LoadBlockQueue {
     Status add_load_id(const UniqueId& load_id);
     void remove_load_id(const UniqueId& load_id);
     void cancel(const Status& st);
+    Status create_wal(int64_t db_id, int64_t tb_id, int64_t wal_id, const std::string& import_label,
+                      WalManager* wal_manager, std::vector<TSlotDescriptor>& slot_desc,
+                      int be_exe_version);
+    Status append_block(vectorized::Block* block);
```
Comment (mymeiyi) [inline be/src/runtime/group_commit_mgr.h:60]: better to rename `write_wal`

File: be/src/runtime/group_commit_mgr.cpp [hunk 2]
```diff
@@ -85,6 +85,8 @@ Status LoadBlockQueue::get_block(vectorized::Block* block, bool* find_block, boo
         _block_queue.pop_front();
         _all_block_queues_bytes->fetch_sub(fblock->bytes(), std::memory_order_relaxed);
         _single_block_queue_bytes->fetch_sub(block->bytes(), std::memory_order_relaxed);
+        //append block
+        RETURN_IF_ERROR(append_block(block));
```
Comment (mymeiyi) [inline be/src/runtime/group_commit_mgr.cpp:89]: should call `write_wal` in add_block function, not `get_block`

File: be/src/runtime/group_commit_mgr.cpp [hunk 3]
```diff
@@ -447,4 +457,22 @@ Status GroupCommitMgr::get_load_block_queue(int64_t table_id, const TUniqueId& i
     }
     return group_commit_table->get_load_block_queue(instance_id, load_block_queue);
 }
+Status LoadBlockQueue::create_wal(int64_t db_id, int64_t tb_id, int64_t wal_id,
+                                  const std::string& import_label, WalManager* wal_manager,
+                                  std::vector<TSlotDescriptor>& slot_desc, int be_exe_version) {
+    _v_wal_writer = std::make_shared<vectorized::VWalWriter>(
+            db_id, tb_id, txn_id, label, wal_manager, slot_desc, be_exe_version);
+    RETURN_IF_ERROR(_v_wal_writer->init());
+    return Status::OK();
```
Comment (mymeiyi) [inline be/src/runtime/group_commit_mgr.cpp:466]: `return _v_wal_writer->init();` is better

File: be/src/vec/sink/writer/vwal_writer.cpp [hunk 4]
```diff
@@ -43,65 +42,47 @@
...
+    RETURN_IF_ERROR(_wal_writer->append_blocks(std::vector<PBlock*> {&pblock}));
     return Status::OK();
 }
 
-Status VWalWriter::append_block(vectorized::Block* input_block, int64_t num_rows,
-                                int64_t filter_rows, vectorized::Block* block,
-                                OlapTableBlockConvertor* block_convertor,
-                                OlapTabletFinder* tablet_finder) {
-    return write_wal(block_convertor, tablet_finder, block, _state, num_rows, filter_rows);
+Status VWalWriter::append_block(vectorized::Block* block) {
```
Comment (mymeiyi) [inline be/src/vec/sink/writer/vwal_writer.cpp:84]: why add this function and not use `write_wal` directly?

File: be/src/vec/sink/writer/vwal_writer.cpp [hunk 5]
```diff
@@ -43,65 +42,47 @@
...
-    for (auto slot_desc : _output_tuple_desc->slots()) {
-        ss << std::to_string(slot_desc->col_unique_id()) << ",";
+    for (auto slot_desc : _slot_descs) {
+        if (slot_desc.col_unique_id < 0) {
+            continue;
+        }
+        ss << std::to_string(slot_desc.col_unique_id) << ",";
     }
     std::string col_ids = ss.str().substr(0, ss.str().size() - 1);
+    LOG(INFO) << "col_ids:" << col_ids;
```
Comment (mymeiyi) [inline be/src/vec/sink/writer/vwal_writer.cpp:70]: Does this have too many logs?

**Comment by hust-hhb** [comment]
already change the code , take a look plz @mymeiyi

**Pull request merged**